src/backends.jl

```diff
@@ -1,20 +1,42 @@
 abstract type AbstractBackend end
 
 struct JuliaBackend <: AbstractBackend end
-struct CaltechHPC <: AbstractBackend end
+abstract type SlurmBackend <: AbstractBackend end
+struct CaltechHPCBackend <: SlurmBackend end
+struct ClimaGPUBackend <: SlurmBackend end
 
 """
     get_backend()
 
-Determine the appropriate backend using relevant system information.
+Get ideal backend for deploying forward model runs. 
+Each backend is found via `gethostname()`. Defaults to JuliaBackend if none is found.
 """
 function get_backend()
-    backend = JuliaBackend
-    if isfile("/etc/redhat-release") &&
-       occursin("Red Hat", read("/etc/redhat-release", String))
-        backend = CaltechHPC
+    hostname = gethostname()
+    if occursin(r"^hpc-(\d\d)-(\d\d).cm.cluster$", hostname) ||
```

Comment:
I think `match` is more appropriate function to use here

Reply:
Is `!isnothing(match(regex, hostname))` then preferable to `occursin(regex, hostname)`, or is there a better way to use `match`?

Reply:
One might wonder why you used `occursin`. What you want to check is that the hostname matches your regular expression, non that it contains it. 

But what you could do is (schematically):
```julia
HOSTNAMES = Dict("^clima.gps.caltech.edu$" => ClimaGPU, "aaaa" => CaltechHPC, "bbb" => CaltechHPC)
backend = get(HOSTNAMES, first(filter(x -> !isnothing(match(x, gethostname(), keys(HOSTNAMES))))
```
(There's certainly more elegant ways)

Reply:
I believe `occursin(regex,str)` checks that the string matches the regex, but your way is already cleaner. I will incorporate it soon, thank you!

Reply:
Yes, it does, but when it is less clear to readers (I think). Your intent is to match the regular expression and you can make it more explicit by using the function match